File: Easy/No_0035_Search Insert Position/35_SearchInsertPosition.py

#Binary Search
class Solution:
    from typing import Sequence
    def searchInsert(self, nums, target):
        ans = self.bin_search(nums, target)
        return ans


    def bin_search(self,a: Sequence, key:int):
        pl = 0
        pr = len(a) - 1
        while True:
            pc =  (pl + pr) // 2
            if a[pc] == key:
                return pc
            elif a[pc] < key:
                pl = pc +1
            else:
                pr = pc -1
            if pl > pr:
                return pl

File: Easy/No_0035_Search Insert Position/35_SearchInsertPosition.js
//Runtime: 52 ms, faster than 81.73% of JavaScript online submissions for Search Insert Position.
//Memory Usage: 33.8 MB, less than 81.25% of JavaScript online submissions for Search Insert Position.

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    var left = 0;
    var right = nums.length -1
    while (left <= right){
        var mid = Math.floor((left + right) / 2)
        if (nums[mid] === target){
            return mid
        }else if (nums[mid] > target){
            right = mid -1
        }else{
            left = mid + 1
        }
    }
    return left
    
};
